sdks/go/alpha.go

```diff
@@ -98,57 +99,57 @@ func (a *Alpha) GetCounterCount(key string) (int64, error) {
 // IncrementCounter increases a counter by the given nonnegative integer amount.
 // Will execute the increment operation against the current CRD value. Will max at max(int64).
 // Will error if the key was not predefined in the GameServer resource on creation.
-// Returns false if the count is at the current capacity (to the latest knowledge of the SDK),
+// Returns error if the count is at the current capacity (to the latest knowledge of the SDK),
 // and no increment will occur.
 //
 // Note: A potential race condition here is that if count values are set from both the SDK and
 // through the K8s API (Allocation or otherwise), since the SDK append operation back to the CRD
 // value is batched asynchronous any value incremented past the capacity will be silently truncated.
-func (a *Alpha) IncrementCounter(key string, amount int64) (bool, error) {
+func (a *Alpha) IncrementCounter(key string, amount int64) error {
 	if amount < 0 {
-		return false, errors.Errorf("CountIncrement amount must be a positive int64, found %d", amount)
+		return errors.Errorf("CountIncrement amount must be a positive int64, found %d", amount)
 	}
 	_, err := a.client.UpdateCounter(context.Background(), &alpha.UpdateCounterRequest{
 		CounterUpdateRequest: &alpha.CounterUpdateRequest{
 			Name:      key,
 			CountDiff: amount,
 		}})
 	if err != nil {
-		return false, errors.Wrapf(err, "could not increment Counter %s by amount %d", key, amount)
+		return errors.Wrapf(err, "Could not increment Counter %s by amount %d", key, amount)
 	}
-	return true, err
+	return err
 }
 
 // DecrementCounter decreases the current count by the given nonnegative integer amount.
 // The Counter Will not go below 0. Will execute the decrement operation against the current CRD value.
-// Returns false if the count is at 0 (to the latest knowledge of the SDK), and no decrement will occur.
-func (a *Alpha) DecrementCounter(key string, amount int64) (bool, error) {
+// Will error if the count is at 0 (to the latest knowledge of the SDK), and no decrement will occur.
+func (a *Alpha) DecrementCounter(key string, amount int64) error {
 	if amount < 0 {
-		return false, errors.Errorf("CountDecrement amount must be a positive int64, found %d", amount)
+		return errors.Errorf("CountDecrement amount must be a positive int64, found %d", amount)
```

Comment:
```suggestion
		return errors.Errorf("amount must be a positive int64, found %d", amount)
```

Reply:
I have made modifications to the error strings.

Please review the following testing and share your feedback..

```
$ nc -u localhost 7654
GET_COUNTER_COUNT rooms
COUNTER: 1
INCREMENT_COUNTER games 1
ERROR: could not increment Counter games by amount 1: rpc error: code = Unknown desc = not found. games Counter not found
INCREMENT_COUNTER rooms 1
ACK: SUCCESS
GET_COUNTER_COUNT rooms
COUNTER: 2
INCREMENT_COUNTER rooms 100
ERROR: could not increment Counter rooms by amount 100: rpc error: code = Unknown desc = out of range. Count must be within range [0,Capacity]. Found Count: 102, Capacity: 10
INCREMENT_COUNTER rooms -1
ERROR: amount must be a positive int64, found -1
DECREMENT_COUNTER rooms 1
ACK: SUCCESS
GET_COUNTER_COUNT rooms
COUNTER: 1
DECREMENT_COUNTER rooms 2
ERROR: could not decrement Counter rooms by amount 2: rpc error: code = Unknown desc = out of range. Count must be within range [0,Capacity]. Found Count: -1, Capacity: 10
DECREMENT_COUNTER rooms -1 
ERROR: amount must be a positive int64, found -1
DECREMENT_COUNTER room 1
ERROR: could not decrement Counter room by amount 1: rpc error: code = Unknown desc = not found. room Counter not found
SET_COUNTER_COUNT rooms 1000
ERROR: could not set Counter rooms count to amount 1000: rpc error: code = Unknown desc = out of range. Count must be within range [0,Capacity]. Found Count: 1000, Capacity: 10
SET_COUNTER_COUNT rooms 5
ACK: SUCCESS
GET_COUNTER_COUNT rooms
COUNTER: 5
SET_COUNTER_COUNT rooms 51
ERROR: could not set Counter rooms count to amount 51: rpc error: code = Unknown desc = out of range. Count must be within range [0,Capacity]. Found Count: 51, Capacity: 10
SET_COUNTER_COUNT rooms -2
ERROR: could not set Counter rooms count to amount -2: rpc error: code = Unknown desc = out of range. Count must be within range [0,Capacity]. Found Count: -2, Capacity: 10
GET_COUNTER_CAPACITY rooms
CAPACITY: 10
SET_COUNTER_CAPACITY rooms 6
ACK: SUCCESS
SET_COUNTER_CAPACITY rooms -6   
ERROR: could not set Counter rooms capacity to amount -6: rpc error: code = Unknown desc = out of range. Capacity must be greater than or equal to 0. Found Capacity: -6
GET_LIST_CAPACITY players
CAPACITY: 100
SET_LIST_CAPACITY players 1000
ACK: SUCCESS
GET_LIST_CAPACITY players
CAPACITY: 1000
SET_LIST_CAPACITY players 1001
ERROR: could not set List players capacity to amount 1001: rpc error: code = Unknown desc = out of range. Capacity must be within range [0,1000]. Found Capacity: 1001
SET_LIST_CAPACITY players -1
ERROR: could not set List players capacity to amount -1: rpc error: code = Unknown desc = out of range. Capacity must be within range [0,1000]. Found Capacity: -1
APPEND_LIST_VALUE players test0
ERROR: could not get List players: rpc error: code = Unknown desc = already exists. Value: test0 already in List: players
GET_LIST_VALUES players
VALUES: test0,test1,test2
APPEND_LIST_VALUE players test3
ACK: SUCCESS
GET_LIST_VALUES players
VALUES: test0,test1,test2,test3
DELETE_LIST_VALUE players test2
ACK: SUCCESS
GET_LIST_VALUES players
VALUES: test0,test1,test3
DELETE_LIST_VALUE players test22
ERROR: could not get List players: rpc error: code = Unknown desc = not found. Value: test22 not found in List: players
```